tests/pfcwd/test_pfcwd_timer_accuracy.py

```diff
@@ -324,17 +345,27 @@ def retrieve_timestamp(self, pattern):
         Returns:
             timestamp_ms (int): syslog timestamp in ms for the line matching the pattern
         """
-        cmd = "grep \"{}\" /var/log/syslog".format(pattern)
-        syslog_msg_list = self.dut.shell(cmd)['stdout'].split()
         try:
-            timestamp_ms = float(self.dut.shell("date -d \"{}\" +%s%3N".format(syslog_msg_list[3]))['stdout'])
-        except RunAnsibleModuleFail:
-            timestamp_ms = float(self.dut.shell("date -d \"{}\" +%s%3N".format(syslog_msg_list[2]))['stdout'])
-        except Exception as e:
-            logging.error("Error when parsing syslog message timestamp: {}".format(repr(e)))
-            pytest.fail("Failed to parse syslog message timestamp")
+            cmd = "grep \"{}\" /var/log/syslog".format(pattern)
+            syslog_msg = self.dut.shell(cmd)['stdout']
+
+            # Regular expressions for the two timestamp formats
+            regex1 = re.compile(r'^[A-Za-z]{3} \d{2} \d{2}:\d{2}:\d{2}\.\d{6}')
+            regex2 = re.compile(r'^\d{4} [A-Za-z]{3} \d{2} \d{2}:\d{2}:\d{2}\.\d{6}')
```

Comment:
@lipxu is there any reasons why we're switching to regex here?

Reply:
Sorry, @auspham , just saw the comments. sorry for the late response. 
I used regex here because I initially aimed to standardize the handling function by searching for the timestamp and processing it with the same function, however, during UT test, I found the date command does not support the format with the year.  therefore, I reverted to the previous handling process but retained the regex change here. thanks.

Reply:
Maybe we can only search the timestamp without year, let me try